server/src/main/java/org/opensearch/cluster/service/TaskBatcher.java

```diff
@@ -173,6 +186,7 @@ void runIfNotProcessed(BatchedTask updateTask) {
                     return tasks.isEmpty() ? entry.getKey() : entry.getKey() + "[" + tasks + "]";
                 }).reduce((s1, s2) -> s1 + ", " + s2).orElse("");
 
+                taskBatcherListener.onProcessed(toExecute);
```

Comment:
should this statement be after run method?

Reply:
We want perform the throttling on number of tasks which are in queue, if we do it after run then it will take tasks which are getting executed as well into consideration for it. So doing it before run method.

Reply:
The listener method is onProcessed and it would be fundamentally confusing if it is being executed before calling processing.

Reply:
Actually TaskBatcher itself first marks all the tasks as processed and then trigger `run` method. 

I agree with your point, it might cause confusion but we need to reduce the count from pending queue before executing those tasks. In that case, should renaming it to `beforeProcess` will make sense?

Reply:
Yes agree with you. We will need `onTimeout` method as well. Will rename this one to `onBeginProcessing`.